client/javascript/components/general/WindowTitle.js

```diff
@@ -0,0 +1,85 @@
+import {IntlProvider, injectIntl} from 'react-intl';
+import DocumentTitle from 'react-document-title';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import * as i18n from '../../i18n/languages';
+import EventTypes from '../../constants/EventTypes';
+import Size from './Size';
+import TransferDataStore from '../../stores/TransferDataStore';
+
+const METHODS_TO_BIND = ['handleTransferChange'];
+
+
+let Title = ({down, up}) => (
+  <span>
+    &darr; <Size value={down} isSpeed />{' '}
+    &uarr; <Size value={up} isSpeed /> |{' '}
+    Flood
+  </span>
+);
+
+Title = injectIntl(Title);
+
+
+class WindowTitle extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      title: 'Flood',
+    };
+
+    this.title = document.createElement('span');
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentDidMount() {
+    TransferDataStore.listen(
+      EventTypes.CLIENT_TRANSFER_SUMMARY_CHANGE,
+      this.handleTransferChange
+    );
+  }
+
+  componentWillUnmount() {
+    TransferDataStore.unlisten(
+      EventTypes.CLIENT_TRANSFER_SUMMARY_CHANGE,
+      this.handleTransferChange
+    );
+  }
+
+  handleTransferChange() {
+    const {children, intl} = this.props;
+
+    const summary = TransferDataStore.getTransferSummary();
+
+    ReactDOM.render((
+      <IntlProvider locale={intl.locale} messages={i18n[intl.locale]}>
+        <Title down={summary.downRate} up={summary.upRate} />
+      </IntlProvider>
+    ), this.title);
+
+    this.setState({
+      title: this.title.innerText,
+    });
+
+    ReactDOM.unmountComponentAtNode(this.title);
```

Comment:
I'm assuming you chose to create this temporary DOM node and grab its `innerText` to leverage the `Size` component, but unfortunately I don't think this is a good solution. It probably has significant performance implications.

Instead, let's abstract the number formatting logic in `Size` into a utility (and let's not include any internationalization in the utility — maybe the utility can return english strings like `GB` and `MB` to denote the unit, and then we'll map these english units to the necessary translation objects in the component). Then, both the `Size` component and this new component can use the utility for getting the appropriate numerical values. Then the values can be passed through `this.props.intl.formatNumber`, and the units can be mapped to the translation object and then get passed to `this.props.intl.formatMessage`.

Reply:
Your assumptions are absolutely correct. I didn't like this approach either, but decided to submit the PR for discussion anyway. Didn't want to refactor anything without your approval of the idea :)

I think what you've described makes sense. I'll try to implement it soon.